src/store/index.ts

```diff
@@ -54,41 +64,73 @@ export const resultsFilter = ref<ResultsFilter>({
   },
 });
 
-export const addFootprint = (spatialId: number) => {
-  if (footprintIds.value === undefined) {
-    footprintIds.value = [];
+export const addFootprint = async (spatialId: number, region?: boolean) => {
+  let key;
+  let footprint;
+  if (!region) {
+    footprint = (await rgdFootprint(spatialId)).footprint;
+    key = `result_${spatialId}`;
+  } else {
+    footprint = regionsList.value?.find((reg) => reg.id === spatialId)?.footprint;
+    key = `region_${spatialId}`;
+  }
+  if (key && footprint) {
+    visibleFootprints.value = { [key]: footprint, ...visibleFootprints.value };
```

Comment:
Should the order be reversed? I.e., should the new `key` override an existing one with the same value?